### PR TITLE
Additional basic DB2 input validation

### PR DESCRIPTION
**Pull request by gjwatts**
Added the following tests for DB2:
testConnectInvalidDatabase
testQueryBlankDatabase
testQueryBlankUsername
testQueryBlankPassword
testQueryBlankTable
testInvalidTableQuery
testInvalidColumnQuery
testInvalidQuery

and returning DB2Exceptions with SQL error code and SQL state.

Also modified the testTimestamp to handle DBs that don't have nano precision.

### CODE REVIEW

**Comment by gjwatts** [comment]
Had to remove the two Java 11 methods I used by mistake

File: vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java [hunk 1]
```diff
@@ -26,66 +26,155 @@
...
+//			System.out.println(ex.getMessage());
+		}
+	}
+
+	@Test
+	public void testQueryBlankUsername(TestContext ctx) {
+		try {
+			options.setUser("");
+			ctx.fail("Expected a DB2Exception to be thrown");
+		} catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (aguibert) [reply vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java:96]: also applies to a few other cases in this PR

File: vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java [hunk 2]
```diff
@@ -26,66 +26,155 @@
...
+			ctx.assertTrue(ex.getMessage().contains("provided was not found") ||
+					ex.getMessage().contains("The connection was closed by the database server"), 
+					"The SQL error message returned is not correct.  It should have contained \"provided was not found\" or \"The connection was closed by the database server\", but instead it said \"" + ex.getMessage() + "\"");
+			ctx.assertTrue(ex.getErrorCode() == SqlCode.DATABASE_NOT_FOUND ||
+					ex.getErrorCode() == SqlCode.CONNECTION_REFUSED, 
+					"Wrong SQL code received");
+			ctx.assertTrue(ex.getSqlState().equalsIgnoreCase("2E000") ||
+					ex.getSqlState() == SQLState.AUTH_DATABASE_CONNECTION_REFUSED,
+					"Wrong SQL state received");
+//			System.out.println(ex.getMessage());
```
Comment (aguibert) [inline vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java:45]: all of these commented out print statements can be removed

File: vertx-db2-client/src/main/java/io/vertx/db2client/DB2ConnectOptions.java [hunk 3]
```diff
@@ -47,11 +49,11 @@ public static DB2ConnectOptions fromUri(String connectionUri) throws IllegalArgu
     public static final String DEFAULT_HOST = "localhost";
     public static final int DEFAULT_PORT = 50000;
     public static final String DEFAULT_USER = "root";
-    public static final String DEFAULT_PASSWORD = "";
-    public static final String DEFAULT_SCHEMA = "";
+    public static final String DEFAULT_PASSWORD = "INVALID";
+    public static final String DEFAULT_SCHEMA = "INVALID";
```
[the remark this answers is not in the document]
Comment (gjwatts) [reply vertx-db2-client/src/main/java/io/vertx/db2client/DB2ConnectOptions.java:53]: Currently, I am checking for blank values that get passed into:
DB2ConnectOptions.setUser
DB2ConnectOptions.setPassword
DB2ConnectOptions.setDatabase

When DB2ConnectOptions.init runs, it fills in default values for user (root), pwd (blank) and DB (blank).  This causes test failures right away for the last two listed.  I could push the testing for blank values back to the parent (SqlConnectOptions), but they already check for null values (not blank) and I am concerned changing global stuff outside of DB2.  How would you recommend I proceed?

File: vertx-db2-client/src/main/java/io/vertx/db2client/DB2ConnectOptions.java [hunk 4]
```diff
@@ -93,17 +95,29 @@ public DB2ConnectOptions setPort(int port) {
 
     @Override
     public DB2ConnectOptions setUser(String user) {
-        return (DB2ConnectOptions) super.setUser(user);
+    	if (user == null || user.trim().length() < 1) {
```
[the remark this answers is not in the document]
Comment (gjwatts) [reply vertx-db2-client/src/main/java/io/vertx/db2client/DB2ConnectOptions.java:98]: Done

File: vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java [hunk 5]
```diff
@@ -26,66 +26,155 @@
...
+		options.setDatabase("DB_DOES_NOT_EXIST");
+		DB2Connection.connect(vertx, options, ctx.asyncAssertFailure(err -> {
+			ctx.assertTrue(err instanceof DB2Exception, "The error message returned is of the wrong type.  It should be a DB2Exception, but it was of type " + err.getClass().getSimpleName());
+			DB2Exception ex = (DB2Exception) err;
+			ctx.assertTrue(ex.getMessage().contains("provided was not found") ||
+					ex.getMessage().contains("The connection was closed by the database server"), 
+					"The SQL error message returned is not correct.  It should have contained \"provided was not found\" or \"The connection was closed by the database server\", but instead it said \"" + ex.getMessage() + "\"");
+			ctx.assertTrue(ex.getErrorCode() == SqlCode.DATABASE_NOT_FOUND ||
+					ex.getErrorCode() == SqlCode.CONNECTION_REFUSED, 
+					"Wrong SQL code received");
```
[the remark this answers is not in the document]
Comment (gjwatts) [reply vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java:41]: Done

File: vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java [hunk 6]
```diff
@@ -26,66 +26,155 @@
...
+			DB2Exception ex = (DB2Exception) err;
+			ctx.assertTrue(ex.getMessage().contains("Invalid credentials"), "The SQL error message returned is not correct.  It should have contained \"Invalid credentials\", but instead it said \"" + ex.getMessage() + "\"");
+			ctx.assertEquals(SqlCode.INVALID_CREDENTIALS, ex.getErrorCode());
+			ctx.assertEquals(SQLState.NET_CONNECT_AUTH_FAILED, ex.getSqlState());
+//			System.out.println(ex.getMessage());
+		}));
+	}
+
+	@Test
+	@Ignore
```
[the remark this answers is not in the document]
Comment (gjwatts) [reply vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java:76]: Nope, thanks for catching that!

File: vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gjwatts) [reply vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java:96]: Yep, good catch!

File: vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java (hunk 2 above)
Comment (gjwatts) [reply vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java:45]: Done

File: vertx-db2-client/src/main/java/io/vertx/db2client/DB2ConnectOptions.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (gjwatts) [reply vertx-db2-client/src/main/java/io/vertx/db2client/DB2ConnectOptions.java]: I have removed the default user/password/schema values and removed the setting of them in the init method.

**Pull request merged**